test/core/test_time.cpp

```diff
@@ -163,6 +163,14 @@ int main() {
   printf("mktime guesses DST (winter): %d\n", tm_winter.tm_isdst == oldDstWinter);
   printf("mktime guesses DST (summer): %d\n", tm_summer.tm_isdst == oldDstSummer);
 
+  struct tm tm_big = {0};
+  tm_big.tm_year = 292278994;
+  tm_big.tm_mday = 31;
+  tm_big.tm_hour = 23;
+  tm_big.tm_min = 59;
+  tm_big.tm_sec = 59;
+  assert(mktime(&tm_big) == -1);
```

Comment:
Can we we also `assert(errno == EOVERFLOW)` here?

In order to do this I suggest we change the return type of `_mktime_js` to `double` and then handle coverting to an int on the native side return a double in `system/lib/libc/mktime.c` where its easy to set errno if NaN is returned.